### PR TITLE
New methods in Entity [10309]

### PR DESCRIPTION
**Pull request by jparisu**
Merge after #1691 

New method : `get_statuscondition`

New unittest for every API method in `Entity`

### CODE REVIEW

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/3561/)
* Linux aarch64[Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux%20aarch64/751/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/3158/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/3593/)

**Review by imontesino — CHANGES_REQUESTED** [review]

**Review by IkerLuengo — CHANGES_REQUESTED** [review]
Nothing to add to the previous comments

File: include/fastdds/dds/core/Entity.hpp [hunk 1]
```diff
@@ -137,12 +148,14 @@ class Entity
     //! StatusMask with triggered statuses set to 1
     StatusMask status_changes_;
 
+    //! InstanceHandle associated to the Entity
+    StatusCondition status_condition_;
```
[the remark this answers is not in the document]
Comment (jparisu) [reply include/fastdds/dds/core/Entity.hpp:152]: :+1:

File: include/fastdds/dds/core/condition/StatusCondition.hpp [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+    // StatusCondition not implemented.
+
+    /**
+     * @brief Defines the list of communication statuses that are taken into account to determine the trigger_value
+     * @param mask defines the mask for the status
+     * @return true if trigger_value is set to true, false otherwise
+     */
+    RTPS_DllAPI fastrtps::types::ReturnCode_t set_enabled_statuses(
+            const StatusMask& mask);
```
[the remark this answers is not in the document]
Comment (jparisu) [reply include/fastdds/dds/core/condition/StatusCondition.hpp:50]: This is a comment for PR #1691 .
I change it there.

File: include/fastdds/dds/core/condition/WaitSet.hpp [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+#include <fastrtps/fastrtps_dll.h>
+#include <fastrtps/types/TypesBase.h>
+
+namespace eprosima {
+namespace fastdds {
+namespace dds {
+
+/**
+ * @brief The WaitSet class allows an application to wait until one or more of the attached Condition objects
+ * has a trigger_value ofTRUE or else until the timeout expires.
```
[the remark this answers is not in the document]
Comment (jparisu) [reply include/fastdds/dds/core/condition/WaitSet.hpp:34]: Same as the previous one, I change it in #1691 .

File: include/fastdds/dds/core/condition/StatusCondition.hpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (jparisu) [reply include/fastdds/dds/core/condition/StatusCondition.hpp:50]: Finally the signature was correct. The comment is the wrong one.

**Review by IkerLuengo — APPROVED** [review]
LGTM

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/3600/)
* Linux aarch64[Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux%20aarch64/790/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/3189/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/3632/)

**Review by imontesino — APPROVED** [review]
LGTM

**Pull request merged**